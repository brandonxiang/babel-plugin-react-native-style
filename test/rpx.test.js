const pluginTester = require('babel-plugin-tester');
const reactNativeStyle = require('../src/index');

pluginTester({
  plugin: reactNativeStyle,
  pluginName: 'babel-plugin-react-native-style',

  tests: [
    {
      snapshot: true,
      title: 'rpx convert',
      pluginOptions: { rpx: { enable: true, size: 750 } },
      code: 'const styles = stylesheet.create({ padding: 1 })',
    },
    {
      snapshot: true,
      title: 'specific rpx',
      pluginOptions: { rpx: { enable: false, size: 750 } },
      code: `
        function call() { return 1 };
        const height = call() + '100rpx'+ 30;
        const styles = stylesheet.create({ padding: "1rpx", height })
      `,
    },
  ],
});
