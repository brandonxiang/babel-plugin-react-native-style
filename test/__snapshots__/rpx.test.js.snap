// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-native-style rpx convert: rpx convert 1`] = `
"
const styles = stylesheet.create({ padding: 1 })

      ↓ ↓ ↓ ↓ ↓ ↓

import { __RPX } from \\"babel-plugin-react-native-style/src/runtime/index\\";
const styles = stylesheet.create({ padding: __RPX(1, 750) });
"
`;
